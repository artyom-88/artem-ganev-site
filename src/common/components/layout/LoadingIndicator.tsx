import { ReactElement } from 'react';
import useStyles from './LoadingIndicator.styles';

const LoadingIndicator = (): ReactElement => {
  const classes = useStyles();
  const className = `${classes.loadingIndicatorContainer} ag-flexbox ag-justifyContent_center ag-alignItems_center`;
  return (
    <div className={className}>
      <div className={classes.loadingIndicator} />
    </div>
  );
};

export default LoadingIndicator;
