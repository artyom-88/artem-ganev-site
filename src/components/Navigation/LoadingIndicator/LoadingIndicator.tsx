import Box from '@material-ui/core/Box';
import { ReactElement } from 'react';
import useStyles from './LoadingIndicator.styles';

/**
 * Simple loading indicator based on SVG-image
 */
const LoadingIndicator = (): ReactElement => {
  const classes = useStyles();
  return (
    <Box
      alignItems='center'
      className={classes.loadingIndicatorContainer}
      display='flex'
      flexBasis='100%'
      flexGrow={1}
      justifyContent='center'
    >
      <div className={classes.loadingIndicator} />
    </Box>
  );
};

export default LoadingIndicator;
