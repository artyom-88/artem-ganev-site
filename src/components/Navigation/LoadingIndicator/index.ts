import { lazy } from 'react';

const LoadingIndicator = lazy(() => import(/* webpackChunkName: "loading-indicator" */ './LoadingIndicator'));

export default LoadingIndicator;
