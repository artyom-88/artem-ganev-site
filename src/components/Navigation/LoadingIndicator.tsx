import React from "react";
import "./LoadingIndicator.scss";

/**
 * Simple loading indicator based on SVG-image
 */
export default () => (
  <div className="flexBox flexGrow-1 flexShrink-1 justifyContentCenter">
    <div className="nav-loading__indicator" />
  </div>
);
