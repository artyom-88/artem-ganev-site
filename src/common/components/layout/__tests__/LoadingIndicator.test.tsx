import LoadingIndicator from 'common/components/layout/LoadingIndicator';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('LoadingIndicator', () => {
  const component = shallow(<LoadingIndicator />);

  it('Should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
