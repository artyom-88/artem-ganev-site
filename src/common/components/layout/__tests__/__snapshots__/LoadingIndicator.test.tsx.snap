// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator Should match snapshot 1`] = `
<Styled(MuiBox)
  alignItems="center"
  className="makeStyles-loadingIndicatorContainer-1"
  display="flex"
  flexBasis="100%"
  flexGrow={1}
  justifyContent="center"
>
  <div
    className="makeStyles-loadingIndicator-2"
  />
</Styled(MuiBox)>
`;
